doc/cascadia/Cascading-Default-Settings.md

```diff
@@ -0,0 +1,571 @@
+we won't touch keys that have the same values as defaults that are currently in
+the `profiles.json` file. Fortunately though, users should be able to remove
+much of the boilerplate from their `profiles.json` files, and trim it down just
+to their modifications.
+
+### Dynamic Profile Generators Need to be Enabled
+With the current proposal, profiles that are generated by a dynamic profile
+generator _need_ that generator to be enabled for the profile to appear in the
+list of profiles. If the generator isn't enabled, then the important parts of
+the profile (name, commandline) will never be set, and the profile's settings
+from the user settings will be ignored at runtime.
+
+For generators where the generation of profiles might be a lengthy process, this
+could negatively impact startup time. Take for example, some hypothetical
+generator that needs to make web requests to generate dynamic profiles. Because
+we need the finalized settings to be able to launch the terminal, we'll be stuck
+loading until that generator is complete.
+
+However, if the user disables that generator entirely, we'll never display that
+profile to the user, even if they've done that setup before.
+
+So the trade-off with this design is that non-existent dynamic profiles will
+never roam to machines where they don't exist and aren't valid, but the
+generators _must_ be enabled to use the dynamic profiles.
+
+## Future considerations
+* It's possible that a very similar layering loading mechanism could be used to
+  layer per-machine settings with roaming settings. Currently, there's only one
+  settings file, and it roams to all your devices. This could be problematic,
+  for example, if one of your machines has a font installed, but another
+  doesn't. A proposed solution to that problem was to have both roaming settings
+  and per-machine settings. The code to layer settings from the defaults and the
+  user settings could be re-used to handle layer the roaming and per-machine
+  settings.
+* What if an extension wants to generate their own dynamic profiles? We've
+  already outlined a contract that profile generators would have to follow to
+  behave correctly. It's possible that we could abstract our implementation into
+  a WinRT interface that extensions could implement, and be triggered just like
+  other dynamic profile generators.
+  - Additionally, we could expand the dynamic profile generator contract to
+    force it to use a "namespace GUID". We'd then have the generator ask us to
+    generate a profile GUID for some static string, given it's namespace GUID.
+    The generator then wouldn't need to always generate unique GUIDs for its
+    profiles, but it would need to generate unique strings. We already allow
+    the user to disable profile generators solely by namespace guid, so they
+    could disable the generator from a certain extension easily.
+* **Multiple settings files** - This could enable us to place color schemes into

```

Comment:
Or you could just concatenate them all. Or have a import thing like vcxprojs do.

Reply:
concat them all requires ordering. not a problem, but if we go that route it needs callout

Reply:
Yea I think it's a complicated enough scenario, which is why I left it out for now.